api/envoy/api/v2/rds.proto

```diff
@@ -108,6 +108,18 @@ message RouteConfiguration {
   // routed by the HTTP connection manager.
   repeated string request_headers_to_remove = 8;
 
+  // By default, headers that should be added/removed are evaluated from most to least specific:
+  //
+  // * route level
+  // * virtual host level
+  // * connection manager level
+  //
+  // To allow setting overrides at the route or virtual host level, this order can be reversed
+  // by setting this option to true. Defaults to false.
+  //
+  // Note: in the v3 API, this will default to true.
```

Comment:
Can you make this a `next-major-version` comment tag (see elsewhere).

Also, if we are going to switch the default should this use `BoolValue`?

Reply:
Done.